lib/shoryuken/middleware/server/exponential_backoff_retry.rb

```diff
@@ -12,14 +12,19 @@ def call(worker, queue, sqs_msg, body)
 
           started_at = Time.now
           yield
-        rescue
+        rescue => ex
+          logger.info { "Message #{sqs_msg.message_id} will attempt retry due to error: #{ex.message}" }
+
           retry_intervals = worker.class.get_shoryuken_options['retry_intervals']
 
           if retry_intervals.nil? || !handle_failure(sqs_msg, started_at, retry_intervals)
             # Re-raise the exception if the job is not going to be exponential backoff retried.
             # This allows custom middleware (like exception notifiers) to be aware of the unhandled failure.
             raise
           end
+
+          # since we didn't raise, lets log the backtrace for debugging purposes.
+          logger.debug { ex.backtrace * "\n  " }
```

Comment:
Is it possible to get into the `rescue` block without an error class being raised?  I don't mind putting in the check, but just curious if the scenario is legitimate or not.